tests/macos/xcodeproj/Test-Target-With-Test-Host-Project.xcodeproj/bazelinstallers/install.sh

```diff
@@ -24,10 +24,14 @@ com.apple.product-type.framework.static)
     # We only care about the entry ending with TARGET_NAME" so that we can get the path to its directory
     QUERY=$(grep -A 2 important_output "$BAZEL_BUILD_EVENT_TEXT_FILENAME" | grep -w uri | grep ${TARGET_NAME}\" | sed "s/uri: \"file:\/\///")
     if [[ -z $QUERY ]]; then
-        # For virtual frameworks, this is disabled
-        exit 0
+        # For virtual frameworks, this is expected to be empty. Nothing to do.
```

Comment:
1. At least as far as I was able to debug this I don't think we can easily (without `xcbuildkit`) prevent `XCBBuildService` from creating it but I could be wrong and missed something here. Still I'd rather make the legacy generator work at some capacity instead of continuing to fix something that is going away soon (see [comment](https://github.com/bazel-ios/rules_ios/pull/661#discussion_r1122168280))

2. My concern with post build is cleaning up more things than necessary and end up removing post-build outputs that we actually need, I figured a pre-build cleanup step would be more reliable to clean what `XCBuildService` did on Xcode launch. No strong opinions here though.

Reply:
Ah so basically you're hitting breakages from the build service creating them? I think it's still possible for it to break then right?

Reply:
If the build service creates these under DD at different times it's possible to hit the indexer failing issues, yes. Then another build should fix it and clean things up again. Based on my testing though after the cleanup happened I navigated to different sources, edited files, changed schemes back and forth and DD continued to be in a good state. If there's something else that triggers the build service re-creating those it's not something that happen often. That's why I'm calling things a workaround, I don't have evidence that this fixes it for good (and it was not the intent).

Reply:
Yeah, I don't know exactly what part of the stack is creating this. Noticed only that on launch `XCBBuildService` is the process responsible for creating the fmw files on disk under DD (without triggering a build in Xcode). It does that most likely based on the project configuration and declared targets/deps.

>When the product type was a .framework we wouldn't install - this used to be good enough in Xcode 13

FWIW the old behaviour should continue to exist, the `QUERY` we had for static fmws should continue to return whatever it was returning before, I just added a conditional to not exit early and instead continue and run `index-import` at the end, we still need that to run even if `input_options` is empty. As long as DD is cleaned up (i.e. not malformed fmw directories in there) the indexing invocations should respect the VFS and compile succesfully.